e2e/snapshot_test.go

```diff
@@ -0,0 +1,214 @@
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Eventually(func() error {
+
+			By("starting create pod with pvc")
+			_, _, err := kubectlWithInput([]byte(pvcYAML), "-n", namespace, "apply", "-f", "-")
+			if err != nil {
+				fmt.Println(err.Error())
+				return err
+			}
+			result, stderr, err := kubectl("-n", namespace, "get", "pod", "snapshot-test", "-o=json")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			By("checking pod is ready")
+			var snapshotPod corev1.Pod
+			err = json.Unmarshal(result, &snapshotPod)
+			if err != nil {
+				return err
+			}
+			err = checkPodReady(&snapshotPod)
+			if err != nil {
+				return err
+			}
+
+			By("writing some data to pvc before snapshot")
+			result, stderr, err = kubectl("-n", namespace, "exec", "snapshot-test", "--", "bash", "-c", "echo hello >> /test1/test.txt")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			By("create snapshot")
+			result, stderr, err = kubectlWithInput([]byte(volumeSnapshotYAML), "-n", namespace, "apply", "-f", "-")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			By("checking snapshot is ready")
+			result, stderr, err = kubectl("-n", namespace, "get", "volumesnapshot", "new-snapshot-test", "-o=json")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			var s snapapi.VolumeSnapshot
+			err = json.Unmarshal(result, &s)
+			if err != nil {
+				return err
+			}
+
+			if s.Status != nil {
+				if s.Status.ReadyToUse == nil {
+					return errors.New("volume snapshot is not yet ready")
+				} else {
+					if !(*s.Status.ReadyToUse) {
+						return errors.New("volume snapshot is not yet ready")
+					}
+				}
+			} else {
+				return errors.New("volume snapshot is not yet ready")
+			}
+
+			By("writing some data to pvc after snapshot")
+			result, stderr, err = kubectl("-n", namespace, "exec", "snapshot-test", "--", "bash", "-c", "echo world >> /test1/test.txt")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			By("restore snapshot")
+			result, stderr, err = kubectlWithInput([]byte(restoreYAML), "-n", namespace, "apply", "-f", "-")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			By("checking restore snapshot pod ready")
+
+			result, stderr, err = kubectl("-n", namespace, "get", "pvc", "myclaim-restore", "-o=json")
+			if err != nil {
+				return fmt.Errorf("%v: stdout=%s, stderr=%s", err, result, stderr)
+			}
+
+			var pvc corev1.PersistentVolumeClaim
+			err = json.Unmarshal(result, &pvc)
+			if err != nil {
+				return err
+			}
+			if pvc.Status.Phase != corev1.ClaimBound {
+				return fmt.Errorf("pvc %s phase is %s but expect %s and status: %s", pvc.Name, pvc.Status.Phase, corev1.ClaimBound, pvc.Status.String())
+			}
```

Comment:
It looks like CI is failing in the same place. I think it is better to output more logs temporary to find out why it is failing.

Reply:
@little-guy-lxr We have run the CI several times, but it keeps failing as @llamerada-jp said. Please investigate the problem on your side and make it stable.  Thank you!